src/bankid.ts

```diff
@@ -308,6 +489,122 @@ export class BankIdClient {
     );
   }
 
+  payment(parameters: PaymentRequest): Promise<PaymentResponse> {
+    if (!parameters.endUserIp || !parameters.userVisibleTransaction) {
+      throw new Error(
+        "Missing required arguments: endUserIp, userVisibleTransaction.",
+      );
+    }
+    if (
+      parameters.userVisibleDataFormat != null &&
+      parameters.userVisibleDataFormat !== "simpleMarkdownV1" &&
+      parameters.userVisibleDataFormat !== "plaintext"
+    ) {
+      throw new Error("userVisibleDataFormat can only be plaintext or simpleMarkdownV1.");
+    }
+
+    parameters = {
+      ...parameters,
+      userVisibleData: parameters.userVisibleData
+        ? Buffer.from(parameters.userVisibleData).toString("base64")
+        : undefined,
+      userNonVisibleData: parameters.userNonVisibleData
+        ? Buffer.from(parameters.userNonVisibleData).toString("base64")
+        : undefined,
+    };
+
+    return this.#call<PaymentRequest, PaymentResponse>(
+      BankIdMethod.payment,
+      parameters,
+    );
+  }
```

Comment:
We are reassigning function argument here.

Maybe we can move it to a new variable (e.g. request / payload) instead? 

```suggestion
    const payload = {
      ...parameters,
      userVisibleData: parameters.userVisibleData
        ? Buffer.from(parameters.userVisibleData).toString("base64")
        : undefined,
      userNonVisibleData: parameters.userNonVisibleData
        ? Buffer.from(parameters.userNonVisibleData).toString("base64")
        : undefined,
    };

    return this.#call<PaymentRequest, PaymentResponse>(
      BankIdMethod.payment,
      payload,
    );
  }
```

Reply:
Same applies for all new bankid methods we are adding.